vm_phys.c

```diff
@@ -0,0 +1,211 @@
+#include <vm_phys.h>
+#include <queue.h>
+#include <kmem.h>
+#include <libkern.h>
+
+/* Note that generally size of page isn't constant
+ * This constant is minimal granularity of page */
+
+#define PAGE_SIZE 4096
+#define NUM_OF_PAGES 128
+#define STACK_SPACE 4 /* stack space in number of pages */
+
+#define ORDER_SIZE(x) (1 << (x))
+
+#define FAIL(MSG...) do { kprintf(MSG); while(1); } while (0);
```

Comment:
Please define a panic function in separate module. Have a look at: [panic.c](http://fxr.watson.org/fxr/source/boot/common/panic.c)